### PR TITLE
[SPARK-25711][Core] Improve start-history-server.sh: show usage User-Friendly and remove deprecated options

### PR DESCRIPTION
**Pull request by gengliangwang**
## What changes were proposed in this pull request?

Currently, if we try run
```
./start-history-server.sh -h
```
We will get such error
```
java.io.FileNotFoundException: File -h does not exist
```

1. This is not User-Friendly.  For option `-h` or `--help`, it should be parsed correctly and show the usage of the class/script.
2. We can remove deprecated options for setting event log directory through command line options.

After fix, we can get following output:
```
Usage: ./sbin/start-history-server.sh [options]


Options:
  --properties-file FILE      Path to a custom Spark properties file.
                              Default is conf/spark-defaults.conf.

Configuration options can be set by setting the corresponding JVM system property.
History Server options are always available; additional options depend on the provider.

History Server options:

  spark.history.ui.port              Port where server will listen for connections
                                     (default 18080)
  spark.history.acls.enable          Whether to enable view acls for all applications
                                     (default false)
  spark.history.provider             Name of history provider class (defaults to
                                     file system-based provider)
  spark.history.retainedApplications Max number of application UIs to keep loaded in memory
                                     (default 50)
FsHistoryProvider options:

  spark.history.fs.logDirectory      Directory where app logs are stored
                                     (default: file:/tmp/spark-events)
  spark.history.fs.updateInterval    How often to reload log data from storage
                                     (in seconds, default: 10)


```

## How was this patch tested?

Manual test

### CODE REVIEW

File: sbin/start-history-server.sh [hunk 1]
```diff
@@ -28,7 +28,22 @@ if [ -z "${SPARK_HOME}" ]; then
   export SPARK_HOME="$(cd "`dirname "$0"`"/..; pwd)"
 fi
 
+# NOTE: This exact class name is matched downstream by SparkSubmit.
+# Any changes need to be reflected there.
+CLASS="org.apache.spark.deploy.history.HistoryServer"
+
+if [[ "$@" = *--help ]] || [[ "$@" = *-h ]]; then
+  echo "Usage: ./sbin/start-history-server.sh [options]"
```
Comment (jiangxb1987) [inline sbin/start-history-server.sh]: nit: why not have a separated `usage()` function?

File: core/src/main/scala/org/apache/spark/deploy/history/HistoryServerArguments.scala [hunk 2]
```diff
@@ -34,26 +34,25 @@ private[history] class HistoryServerArguments(conf: SparkConf, args: Array[Strin
...
+        parse(tail)
+
+      case ("--help" | "-h") :: tail =>
+        printUsageAndExit(0)
+
+      case ("--properties-file") :: value :: tail =>
+        propertiesFile = value
+        parse(tail)
+
+      case dir :: Nil =>
```
Comment (jiangxb1987) [inline core/src/main/scala/org/apache/spark/deploy/history/HistoryServerArguments.scala]: IIUC this is not related to the PR description?

File: sbin/start-history-server.sh (hunk 1 above)
Comment (gengliangwang) [reply sbin/start-history-server.sh]: Well this is short, and I am following what `start-master.sh` and `start-slave.sh` did.

File: core/src/main/scala/org/apache/spark/deploy/history/HistoryServerArguments.scala (hunk 2 above)
Comment (gengliangwang) [reply core/src/main/scala/org/apache/spark/deploy/history/HistoryServerArguments.scala]: No, it is related. For the last single argument, it is treated as the location of directory.
See the deleted code
```
if (args.length == 1) {
  setLogDirectory(args.head)
}
```
I prefer to keep the behavior.

Comment (jiangxb1987) [reply core/src/main/scala/org/apache/spark/deploy/history/HistoryServerArguments.scala]: I'm not against the change, but we shall mention it in the PR desc.

File: sbin/start-history-server.sh (hunk 1 above)
Comment (jiangxb1987) [reply sbin/start-history-server.sh]: Well, I also saw similar code in `start-thriftserver.sh`, it uses `usage()`. Both are fine to me, just head up to make sure we've taken that into consideration.

File: core/src/main/scala/org/apache/spark/deploy/history/HistoryServerArguments.scala (hunk 2 above)
Comment (jiangxb1987) [reply core/src/main/scala/org/apache/spark/deploy/history/HistoryServerArguments.scala]: sounds good.

Comment (gengliangwang) [reply core/src/main/scala/org/apache/spark/deploy/history/HistoryServerArguments.scala]: I see. I have updated the description.

**Review by srowen — APPROVED** [review]
Looks OK but update the title. It's not clear that it's about showing a help message on the command line

**Comment by dongjoon-hyun** [comment]
BTW, @srowen , @jiangxb1987 , and @gengliangwang .

Spark 3.0 is a good chance to remove the deprecated options. Shall we remove the following? If we don't this now, this can happen in Spark 4.0.
```
  DIR                         Deprecated; set spark.history.fs.logDirectory directly
  --dir DIR (-d DIR)          Deprecated; set spark.history.fs.logDirectory directly
```

**Comment by srowen** [comment]
Agree that's a good idea

**Comment by gengliangwang** [comment]
Agree. Let me remove them in this PR.

**Comment by jiangxb1987** [comment]
Let's also update the title to include the deprecation changes.

**Comment by gengliangwang** [comment]
retest this please.

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by gengliangwang** [comment]
retest this please.

**Comment by gengliangwang** [comment]
retest this please.

**Review by srowen — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Merged to master. Thank you,  @gengliangwang , @srowen , @jiangxb1987 .
